File: app/src/main/java/com/gallosalocin/go4lunch/services/RestaurantApi.java
package com.gallosalocin.go4lunch.services;

import com.gallosalocin.go4lunch.services.dto.DetailsResponse;
import com.gallosalocin.go4lunch.services.dto.RestaurantResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface RestaurantApi {

    @GET("nearbysearch/json?")
    Call<RestaurantResponse> getNearbyRestaurant(@Query("location") String location,
                                                 @Query("radius") int radius,
                                                 @Query("type") String type,
                                                 @Query("key") String key);

    @GET("details/json?")
    Call<DetailsResponse> getDetailsRestaurant(@Query("place_id") String placeId,
                                               @Query("key") String key);
}

File: app/src/main/java/com/gallosalocin/go4lunch/services/RestaurantApi.kt
package com.gallosalocin.go4lunch.services

import com.gallosalocin.go4lunch.services.dto.DetailsResponse
import com.gallosalocin.go4lunch.services.dto.RestaurantResponse
import io.reactivex.rxjava3.core.Observable
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface RestaurantApi {

    @GET("nearbysearch/json?")
    fun getNearbyRestaurant(
            @Query("location") location: String,
            @Query("radius") radius: Int,
            @Query("type") type: String,
            @Query("key") key: String
    ): Observable<RestaurantResponse>

    @GET("details/json?")
    fun getDetailsRestaurant(
            @Query("place_id") placeId: String,
            @Query("key") key: String
    ): Call<DetailsResponse>
}